addons/product/pricelist.py

```diff
@@ -453,23 +298,48 @@ def _check_margin(self, cr, uid, ids, context=None):
                 return False
         return True
 
+    def _compute_pricelist_name(self, cr, uid, ids, name, args, context=None):
+        for record in self.browse(cr, uid, ids, context=context):
+            if record.compute_price == 'fixed':
+                res = str(record.fixed_price) + " " + record.currency_id.name
+            elif record.compute_price == 'percentage':
+                res = str(record.percent_price) + " % discount "
+            else:
+                res = str(abs(record.price_discount)) + " % discount + " + str(record.price_surcharge) + " surcharge "
+        return res
+
+    def _price_list_item_name(self, cr, uid, ids, name, args, context=None):
+        res = {}
+        for record in self.browse(cr, uid, ids, context=context):
+            res[record.id] = {'name': '', 'price_name': ''}
+            if record.categ_id:
+                res[record.id]['name'] = " Category : " + record.categ_id.name
+            elif record.product_tmpl_id:
+                res[record.id]['name'] = record.product_tmpl_id.name
+            elif record.product_id:
+                product_name = self.pool['product.product'].name_get(cr, uid, [record.product_id.id], context=context)[0][1]
+                product_name = product_name.replace('[%s]' % record.product_id.code, '')
+                res[record.id]['name'] = product_name
+            else:
+                res[record.id]['name'] = "All Products"
+            res[record.id]['price_name'] = self._compute_pricelist_name(cr, uid, ids, name, args, context=context)
```

Comment:
what is purpose of this stuff ?

Please check argument : ids. this statement will be execute inside loop. so particular one id should pass instead of all ids...